conformance/tests/httproute-request-redirect.yaml

```
@@ -0,0 +1,40 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: request-redirect
+  namespace: gateway-conformance-infra
+spec:
+  parentRefs:
+  - name: same-namespace
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /hostname
```

For some reason "hostname" as the path matcher feels a bit confusing to me, maybe a slight tweak would make it clearer.
```suggestion
        value: /hostname-redirect
```